### PR TITLE
externals: allow user to use system Vulkan headers

### PR DESCRIPTION
**Pull request by Castor215**
This change is [Reviewable](https://reviewable.io/reviews/citra-emu/citra/7155)

### CODE REVIEW

**Comment by Wunkolo** [comment]
Thanks for this.
This would address the many `VULKAN_HPP_ASSERT( d.getVkHeaderVersion() == VK_HEADER_VERSION );`-asserts I would hit at times from having the vulkan-hpp submodule and my system's newer vulkan-hpp seeming to be used at the same time.

File: externals/CMakeLists.txt [hunk 1]
```diff
@@ -376,9 +376,17 @@ endif()
...
+    find_package(Vulkan REQUIRED)
+    if(TARGET Vulkan::Headers)
+        message(STATUS "Found Vulkan headers")
+        target_link_libraries(vulkan-headers INTERFACE Vulkan::Headers)
+    endif()
+else()
+    target_include_directories(vulkan-headers SYSTEM INTERFACE ./vulkan-headers/include)
+    if (APPLE)
+        target_include_directories(vulkan-headers SYSTEM INTERFACE ${CMAKE_CURRENT_SOURCE_DIR}/MoltenVK)
+    endif()
```
Comment (Steveice10) [inline externals/CMakeLists.txt:389]: I believe this should be separate as these are not part of the general system Vulkan headers.

**Comment by Steveice10** [comment]
Linux build issue is due to a caching problem I'm fixing so don't worry about it.

**Review by Steveice10 — APPROVED** [review]

File: externals/CMakeLists.txt (hunk 1 above)
Comment (Castor215) [reply externals/CMakeLists.txt]: Fixed

**Review by FearlessTobi — APPROVED** [review]

**Pull request merged**